draco/asp/helpers.lp

```diff
@@ -60,10 +60,28 @@ attribute(mark_channel_discrete_or_binned,M,C) :-
     attribute((encoding,channel),E,C),
     attribute((encoding,binning),E,_).
 
+% @helper(mark_encoding_discrete_or_binned) Whether a mark has a discrete scale or binned encoding for an encoding.
+attribute(mark_encoding_discrete_or_binned,M,E) :-
+    entity(encoding,M,E),
+    attribute((encoding,channel),E,C),
+    attribute(mark_scale_channel,M,T,C),
+    domain(discrete_scale,T).
+attribute(mark_encoding_discrete_or_binned,M,E) :-
+    entity(encoding,M,E),
+    attribute((encoding,binning),E,_).
+
 % @helper(encoding_cardinality,E,N) The cardinality of an encoding.
 attribute(encoding_cardinality,E,N) :-
     attribute((field,unique),F,N),
     attribute((encoding,field),E,F),
     not attribute((encoding,binning),E,_).
```

Comment:
I think this also need to be discrete?

Reply:
considering how it's used in `@soft(horizontal_scrolling_x)`, etc.

the facet_cardinality helper might have the same problem